test/integration/endpoints/endpoints_test.go

```diff
@@ -158,6 +160,146 @@ func TestEndpointUpdates(t *testing.T) {
 
 }
 
+func TestEndpointWithMultiplePodUpdates(t *testing.T) {
```

Comment:
```suggestion
// regression test for https://issues.k8s.io/125638
func TestEndpointWithMultiplePodUpdates(t *testing.T) {
```

This test is great, it reproduces the problem